lib/farscape/agent.rb

```diff
@@ -23,10 +39,19 @@ def representor
       safe? ? SafeRepresentorAgent : RepresentorAgent
     end
 
-    def enter(entry = entry_point)
-      @entry_point ||= entry
+    # Discovers provided a key and template_variables.
+    # This method is here to be easily overwritten or monkey-patched if needed.
+    def discover_entry_point(key, template_variables={})
+      Discovery.new.discover(self.class.config, key, template_variables)
+    end
+
+    def enter(entry = entry_point, template_variables = {})
```

Comment:
Just a ~~stupid~~ question about Ruby. `entry = entry_point`. What `entry_point` does this point to? Is it the `@entry_point`?